pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -1773,6 +1871,27 @@ public static String getCollaboratorSampleId(int kitTypeId, String participantCo
         }
     }
 
+    public static String getParticipantIdFromLegacyKit(DDPInstance ddpInstance, String shortId, String collaboratorParticipantId) {
+        Profile profile = new ElasticSearchService().getParticipantProfileByShortID(shortId, ddpInstance.getParticipantIndexES());
+        if (profile == null) {
+            throw new DsmInternalError("Could not find profile for shortId " + shortId);
+        }
+        if (StringUtils.isBlank(profile.getLegacyAltPid())) {
+            return null;
+        }
+        if (isLegacyCollaboratorParticipantId(collaboratorParticipantId, profile.getLegacyShortId())) {
+            List<KitRequestShipping> legacyKits = kitRequestDao.getKitRequestsForCollaboratorParticipantId(collaboratorParticipantId);
+            if (!legacyKits.isEmpty()) {
+                return legacyKits.get(0).getDdpParticipantId();
+            }
+        }
+        return null;
+    }
+
+    private static boolean isLegacyCollaboratorParticipantId(String collaboratorParticipantId, String legacyShortId) {
```

Comment:
Should this method move to the Profile class, or another ES participant-ish class?  It seems like an attribute of the participant, not a kit request.

Reply:
I think this is more kit related and it is not related to profile  because it's about checking something on a collaborator participant id which is a kit related value. IOW there isn't a specific class to put this in to me